lib/SIL/SILWitnessTable.cpp

```diff
@@ -155,3 +156,18 @@ void SILWitnessTable::convertToDefinition(ArrayRef<Entry> entries,
 Identifier SILWitnessTable::getIdentifier() const {
   return Mod.getASTContext().getIdentifier(Name);
 }
+
+bool SILWitnessTable::conformanceIsSerialized(ProtocolConformance *conformance,
+                                              ResilienceStrategy strategy,
+                                              bool silSerializeWitnessTables) {
+  auto *nominal = conformance->getType()->getAnyNominal();
+  // Only serialize if the witness table is sufficiently static, andresilience
+  // is explicitly enabled for this compilation or if we serialize all eligible
+  // witness tables.
+  return (strategy == ResilienceStrategy::Resilient ||
```

Comment:
These are `&&` so I'm not sure it would be. It'd have to be something like

```swift
if (strategy != ResilienceStrategy::Resilient && !silSerializeWitnessTables)
    return false;

if (!nominal->hasFixedLayout())
    return false;

...

return true;
```

which I personally find much harder to decipher (double negatives...). I can, however, break it up into variables with names.